src/stp/plays/defensive/DefendShot.cpp

```diff
@@ -23,52 +23,118 @@ DefendShot::DefendShot() : Play() {
 
     return flagMap;
 }
 
 void DefendShot::calculateInfoForRoles() noexcept {
+    calculateInfoForWallers();
     calculateInfoForDefenders();
     calculateInfoForHarassers();
     calculateInfoForKeeper();
 }
 
+void DefendShot::calculateInfoForWallers() noexcept{
+    stpInfos["waller_1"].setPositionToDefend(field.getOurTopGoalSide());
+    stpInfos["waller_1"].setBlockDistance(BlockDistance::CLOSE);
+
+    stpInfos["waller_2"].setPositionToDefend(field.getOurGoalCenter());
+    stpInfos["waller_2"].setBlockDistance(BlockDistance::CLOSE);
+
+    stpInfos["waller_3"].setPositionToDefend(field.getOurBottomGoalSide());
+    stpInfos["waller_3"].setBlockDistance(BlockDistance::CLOSE);
+}
+
 void DefendShot::calculateInfoForDefenders() noexcept {
-    stpInfos["defender_1"].setPositionToDefend(field.getOurTopGoalSide());
+    auto enemyRobots = world->getWorld()->getThem();
+
+    auto enemyClosestToBall = world->getWorld()->getRobotClosestToBall(world::them);
+
+    enemyRobots.erase(std::remove_if(enemyRobots.begin(), enemyRobots.end(),
+                                     [&](const auto enemyRobot) -> bool { return enemyClosestToBall && enemyRobot->getId() == enemyClosestToBall.value()->getId(); }),
+                      enemyRobots.end());
+
+    auto enemyClosestToOurGoalOne= world->getWorld()->getRobotClosestToPoint(field.getOurGoalCenter(), enemyRobots);
+
+    enemyRobots.erase(std::remove_if(enemyRobots.begin(), enemyRobots.end(),
+                                     [&](const auto enemyRobot) -> bool { return enemyClosestToOurGoalOne && enemyRobot->getId() == enemyClosestToOurGoalOne.value()->getId(); }),
+                      enemyRobots.end());
+
+    auto enemyClosestToOurGoalTwo = world->getWorld()->getRobotClosestToPoint(field.getOurGoalCenter(), enemyRobots);
+
+    enemyRobots.erase(std::remove_if(enemyRobots.begin(), enemyRobots.end(),
+                                     [&](const auto enemyRobot) -> bool { return enemyClosestToOurGoalTwo && enemyRobot->getId() == enemyClosestToOurGoalTwo.value()->getId(); }),
+                      enemyRobots.end());
+
+    stpInfos["defender_1"].setPositionToDefend(field.getOurGoalCenter());
+    stpInfos["defender_1"].setEnemyRobot(enemyClosestToOurGoalOne);
     stpInfos["defender_1"].setBlockDistance(BlockDistance::CLOSE);
 
-    stpInfos["defender_2"].setPositionToDefend(field.getOurBottomGoalSide());
+
+    stpInfos["defender_2"].setPositionToDefend(field.getOurGoalCenter());
+    stpInfos["defender_2"].setEnemyRobot(enemyClosestToOurGoalTwo);
     stpInfos["defender_2"].setBlockDistance(BlockDistance::CLOSE);
+
+    std::map<double, Vector2> enemyMap;
+
+    for (auto enemy : enemyRobots){
+        double score = FieldComputations::getDistanceToGoal(field, true, enemy->getPos());
+        enemyMap.template insert(std::pair<double, Vector2>(score, enemy->getPos()));
+    }
+
+    for (int i = 1; i <= 4; i++){
```

Comment:
There should be checks to make sure enemyMap is not empty, as that would cause errors when trying to erase an element of the vector.